### PR TITLE
[Kotlin Cleanup] Statistics.kt

### PR DESCRIPTION
**Pull request by allisonbickford**
## Purpose / Description
Fixes some warning from running code analysis on the file.

There are a couple outstanding `@KotlinCleanup` items I'd appreciate if anyone could explain or clarify for me.

_line 85_
```
@KotlinCleanup("cleanup: scope function")
```

_line 133_
```
@KotlinCleanup("cleanup: findItem -> isChecked to nested function")
```

## Fixes
Part of #10489 

## Approach
Removes some `@KotlinCleanup`s and simplifies designated blocks of code

## How Has This Been Tested?
Navigated to "Statistics" section in sidebar (physical device), checked for any irregularities or crashes.

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by david-allison** [comment]
Line 133:

either introduce `fun setMenuItemCheck(@IDRes id: int) { }`

or:

```kotlin
            val itemToCheck = when (taskHandler!!.statType) {
                AxisType.TYPE_MONTH -> R.id.item_time_month
                AxisType.TYPE_YEAR -> R.id.item_time_year
                AxisType.TYPE_LIFE -> R.id.item_time_all
            }
            menu.findItem(itemToCheck).isChecked = true
```

**Comment by david-allison** [comment]
Line 85:

```kotlin
        viewPager = findViewById<ViewPager2?>(R.id.pager).apply {
            adapter = StatsPagerAdapter(this@Statistics)
            offscreenPageLimit = 8
        }
```

**Comment by allisonbickford** [comment]
> Looks great! minor changes requested (`onCollectionLoaded` is a massive architectural pain which we should sort out)

thank you for such helpful comments 🙌🏻 everything you mentioned should be updated now

File: AnkiDroid/src/main/java/com/ichi2/anki/Statistics.kt [hunk 1]
```diff
@@ -72,54 +71,55 @@ class Statistics : NavigationDrawerActivity(), DeckSelectionListener, SubtitleLi
...
+        // Create the adapter that will return a fragment for each of the three
+        // primary sections of the activity.
+        // Set up the ViewPager with the sections adapter.
+        viewPager = findViewById<ViewPager2?>(R.id.pager).apply {
+            adapter = StatsPagerAdapter(this@Statistics)
+            offscreenPageLimit = 8
+        }
+        slidingTabLayout = findViewById(R.id.sliding_tabs)
+        // Setup Task Handler
+        taskHandler = getInstance(col)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/Statistics.kt:84]: This needs to moved back + non lateinit. Can't guarantee access to `col` before `onCollectionLoaded`

**Review by david-allison — CHANGES_REQUESTED** [review]
Sorry, `onCollectionLoaded` is a massive pain

File: AnkiDroid/src/main/java/com/ichi2/anki/Statistics.kt (hunk 1 above)
Comment (allisonbickford) [reply AnkiDroid/src/main/java/com/ichi2/anki/Statistics.kt]: oops, I didn't mean to move that one. should be ✅

**Review by mikehardy — APPROVED** [review]
seems fine but David's been collaborating on this one - curious for his take

**Review by david-allison — APPROVED** [review]
I have way too many notifications, sorry this one got lost!

**Comment by mikehardy** [comment]
Hi there! APRIL OpenCollective Notice

Just a friendly notice that we try to process OpenCollective payments monthly - it's time for [April 2022 submissions](https://github.com/ankidroid/Anki-Android/pulls?q=is%3Apr+merged%3A2022-04-01..2022-04-30)

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

I only post one comment per person to avoid spamming you, regardless of the number of PRs merged, but this note applies to all PRs merged for in the month of March.

Please note that GSoC contributions are okay for this process. Our philosophy is that our users have donated to AnkiDroid for all contributions. The only PRs that will not go through the OpenCollective process are ones directly related to am accepted GSoC project from a selected participant, since those receive a stipend from GSoC itself.

Please understand that this means the monthly budget we have for payments might be stretched a bit and the cap on payments-per-person may be lower, but we try to make our process as fair and transparent as possible, we just need your understanding.

Thanks!

**Pull request merged**